src/main.c

```diff
@@ -604,6 +613,24 @@ int main(int argc, char *argv[])
 				free(cookie);
 				break;
 			}
+			if (strcmp(long_options[option_index].name,
+					"ext-browser-saml") == 0) {
+				long port = 8020;
+				if (optarg != NULL) {
+					port = strtol(optarg, NULL, 0);
+					if (port < 1 || port > 65535) {
+						log_error("Specify a valid listen port or omit for parameter ext-browser-saml\n");
+						goto user_error;
+					}
+				}
+				cli_cfg.listen_port = (uint16_t)port;
+				break;
+			}
+			if (strcmp(long_options[option_index].name, "auth-id") == 0) {
+				free(cli_cfg.auth_id);
+				cli_cfg.auth_id = strdup(optarg);
```

Comment:
I find it handy to have at least a zero length check to catch shell mistakes like empty variables when calling as early as possible.